src/metabase/api/slack.clj

```diff
@@ -10,20 +11,32 @@
 
 (api/defendpoint PUT "/settings"
   "Update Slack related settings. You must be a superuser to do this."
-  [:as {{slack-app-token :slack-app-token} :body}]
-  {slack-app-token     (s/maybe su/NonBlankString)}
+  [:as {{slack-app-token :slack-app-token, slack-files-channel :slack-files-channel} :body}]
+  {slack-app-token     (s/maybe su/NonBlankString)
+   slack-files-channel (s/maybe su/NonBlankString)}
   (api/check-superuser)
-  (if-not slack-app-token
-    (slack/slack-app-token nil)
-    (try
-      (when-not config/is-test?
-        (when-not (slack/valid-token? slack-app-token)
-          (throw (ex-info (tru "Invalid Slack token.") {:status-code 400}))))
-      ;; Clear the deprecated `slack-token` when setting a new `slack-app-token`
-      (slack/slack-token nil)
-      (slack/slack-app-token slack-app-token)
-      {:ok true}
-      (catch clojure.lang.ExceptionInfo info
-        {:status 400, :body (ex-data info)}))))
+  (try
+    (when (and slack-app-token (not config/is-test?))
+      (when-not (slack/valid-token? slack-app-token)
+        (throw (ex-info (tru "Invalid Slack token.") {:status-code 400}))))
+    (slack/slack-app-token slack-app-token)
+    (when slack-app-token
+      (do
+        (slack/slack-token-valid? true)
+        ;; Clear the deprecated `slack-token` when setting a new `slack-app-token`
+        (slack/slack-token nil)))
+    (slack/slack-files-channel slack-files-channel)
+    {:ok true}
+    (catch clojure.lang.ExceptionInfo info
+      {:status 400, :body (ex-data info)})))
+
+(def ^:private slack-manifest
+  (delay (slurp (io/resource "slack-manifest.yaml"))))
```

Comment:
probably fine for now. But `jar tf 0.41.2.jar | grep "\.yaml$" | wc -l` shows there are 30 yaml files in our jar. I wonder if at some point we should start putting our resources in a `resources/metabase/metabase` directory so we can ensure no dependency puts a "slack-manifest.yaml" on the classpath

Reply:
Interesting. I can open a tech debt issue to keep track of this.